tools/CheckMetadata.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import datetime
+import git
+import os
+import re
+import unicodedata
+import unittest
+import urllib
+import yaml
+
+VERSION_PATTERN = r'(\d{4})\.(\d{2})\.(\d{2})(\.\d+)?'
+DOI_PATTERN = r'10\.\d{4,9}/zenodo\.(\d+)'
+ORCID_PATTERN = r'\d{4}-\d{4}-\d{4}-\d{3}[0-9X]'
```

Comment:
I haven't look, but are we guaranteed ORCID won't add more letters? If not, maybe in the check below add to the error message that it could be that more letters should be added?

Reply:
Yea the letter is just the checksum:

https://support.orcid.org/hc/en-us/articles/360006897674-Structure-of-the-ORCID-Identifier

> The last character in the ORCID iD is a checksum. In accordance with ISO/IEC 7064:2003, MOD 11-2, this checksum must be "0" - "9" or "X", a capital letter X which represents the value 10.